pkg/kubefed2/join.go

```diff
@@ -534,6 +552,116 @@ func createClusterRoleBinding(clusterClientset client.Interface, saName, namespa
 	return &clusterRoleBinding, nil
 }
 
+// createRoleBinding creates an RBAC role and binding that allows the
+// service account identified by saName to access all resources in the
+// specified namespace.
+func createRoleBinding(clusterClientset client.Interface, saName, namespace,
+	joiningClusterName string, dryRun bool) (*rbacv1.RoleBinding, error) {
+	roleName := util.RoleName(saName)
+
+	rules := []rbacv1.PolicyRule{
+		{
+			Verbs:     []string{rbacv1.VerbAll},
+			APIGroups: []string{rbacv1.APIGroupAll},
+			Resources: []string{rbacv1.ResourceAll},
+		},
+		{
+			Verbs:           []string{"Get"},
+			NonResourceURLs: []string{"/healthz"},
+		},
+	}
+
+	role := &rbacv1.Role{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: roleName,
```

Comment:
It is better to add namespace here for metadata. I know that it is not required, but adding namespace can make the logic clear.

Reply:
There is a single namespace parameter in this function, I'm not sure how adding it redundantly to the type is going to 'make the logic clear'.

Reply:
Usually for a Role, we always have `namespace` specified in `metadata` as follows:

```
kind: Role
apiVersion: rbac.authorization.k8s.io/v1
metadata:
  namespace: default
  name: pod-reader
rules:
- apiGroups: [""] # "" indicates the core API group
  resources: ["pods"]
  verbs: ["get", "watch", "list"]
```

For ClusterRole, do not specify namespace.
```
kind: ClusterRole
apiVersion: rbac.authorization.k8s.io/v1
metadata:
  # "namespace" omitted since ClusterRoles are not namespaced
  name: secret-reader
rules:
- apiGroups: [""]
  resources: ["secrets"]
  verbs: ["get", "watch", "list"]
```

https://kubernetes.io/docs/reference/access-authn-authz/rbac/#api-overview

Reply:
I get that it's possible to specify the namespace for a namespaced-scoped resource, and not possible for a cluster-scoped resource.  It's not necessary to specify the namespace for a namespace-scoped resource where the namespace will be specified in a creation call (e.g. `kubectl -n foo create -f myresource.yaml` or `MyResources("foo").Create()`).  

I disagree that there is value in including it 'for clarity' in this case, since the definition and creation of the resource are both performed in the context of a simple function.